dfastmi/batch/AreaDetector.py

```diff
@@ -0,0 +1,389 @@
+License along with this library; if not, see <http://www.gnu.org/licenses/>.
+
+contact: delft3d.support@deltares.nl
+Stichting Deltares
+P.O. Box 177
+2600 MH Delft, The Netherlands
+
+All indications and logos of, and references to, "Delft3D" and "Deltares"
+are registered trademarks of Stichting Deltares, and remain the property of
+Stichting Deltares. All rights reserved.
+
+INFORMATION
+This file is part of D-FAST Morphological Impact: https://github.com/Deltares/D-FAST_Morphological_Impact
+"""
+import math
+from typing import Tuple
+
+import numpy
+
+
+class AreaDetector:
+
+    @property
+    def area(self) -> numpy.ndarray:
+        """
+        area
+        """
+        return self._area
+
+    @property
+    def volume(self) -> numpy.ndarray:
+        """
+        area volume
+        """
+        return self._volume
+
+    @property
+    def area_list(self) -> list:
+        """
+        List of sub areas
+        """
+        return self._area_list
+
+    @property
+    def total_area_weigth(self) -> numpy.ndarray:
+        """
+        total area weigth
+        """
+        return self._total_area_weigth
+
+    def __init__(self):
+        self._area: numpy.ndarray = numpy.zeros(0)
+        self._volume: numpy.ndarray = numpy.zeros(0)
+        self._area_list: list = []
+        self._total_area_weigth: numpy.ndarray = numpy.zeros(0)
+
+    def detect_areas(
+        self,
+        dzgemi: numpy.ndarray,
+        dzmin: float,
+        edgeface_indeces: numpy.ndarray,
+        areai: numpy.ndarray,
+        wbin: numpy.ndarray,
+        wthresh: numpy.ndarray,
+        siface: numpy.ndarray,
+        afrac: numpy.ndarray,
+        sbin: numpy.ndarray,
+        sthresh: numpy.ndarray,
+        slength: float,
+    ):
+        self._total_area_weigth = numpy.zeros(dzgemi.shape)
+
+        sbin_length = sthresh[1] - sthresh[0]
+        nwidth = wthresh[-1] - wthresh[0]
+        sub_areai, n_sub_areas = self._detect_connected_regions(
+            dzgemi > dzmin, edgeface_indeces
+        )
+        print("number of areas detected: ", n_sub_areas)
+
+        area = numpy.zeros(n_sub_areas)
+        volume = numpy.zeros((3, n_sub_areas))
+        sub_area_list = []
+
+        for ia in range(n_sub_areas):
+            dzgemi_filtered = dzgemi.copy()
+            dzgemi_filtered[sub_areai != ia] = 0.0
+            sub_area_list.append(sub_areai == ia)
+
+            volume[1, ia], wght_area_ia = self._comp_sedimentation_volume1(
+                dzgemi_filtered,
+                dzmin,
+                areai,
+                wbin,
+                siface,
+                afrac,
+                sbin,
+                wthresh,
+                slength,
+                sbin_length,
+            )
+            self._total_area_weigth += wght_area_ia
+
+            volume[2, ia], area[ia], volume[0, ia] = self._comp_sedimentation_volume2(
+                numpy.maximum(dzgemi_filtered, 0.0), dzmin, areai, slength, nwidth
```

Comment:
Is `nwidth` a float?

Reply:
I see nwidth = wthresh[-1] - wthresh[0],  since wtrash is an ndarray, nwidth could basicly be anything 💀

Reply:
assuming nwidth stands for normal_width, which is also know in reach as a float I assume it should be float.
Based on the information in this method it could be anything.

Reply:
checked together, now tpehinted as float.